clients/src/main/java/org/apache/kafka/common/requests/UpdateFeaturesResponse.java

```diff
@@ -82,17 +84,31 @@ public static UpdateFeaturesResponse parse(ByteBuffer buffer, short version) {
         return new UpdateFeaturesResponse(new UpdateFeaturesResponseData(new ByteBufferAccessor(buffer), version));
     }
 
-    public static UpdateFeaturesResponse createWithErrors(ApiError topLevelError, Map<String, ApiError> updateErrors, int throttleTimeMs) {
+    public static UpdateFeaturesResponse createWithErrors(short version, ApiError topLevelError, Map<String, ApiError> updateErrors, int throttleTimeMs) {
         final UpdatableFeatureResultCollection results = new UpdatableFeatureResultCollection();
-        for (final Map.Entry<String, ApiError> updateError : updateErrors.entrySet()) {
-            final String feature = updateError.getKey();
-            final ApiError error = updateError.getValue();
-            final UpdatableFeatureResult result = new UpdatableFeatureResult();
-            result.setFeature(feature)
-                .setErrorCode(error.error().code())
-                .setErrorMessage(error.message());
-            results.add(result);
+        Optional<Map.Entry<String, ApiError>> errorEntry = Optional.empty();
+        if (version > 1) {
+            Stream<Map.Entry<String, ApiError>> errorEntries = updateErrors.entrySet().stream().filter(entry ->
+                    !entry.getValue().error().equals(Errors.NONE));
+            errorEntry = errorEntries.findFirst();
         }
+
+        if (errorEntry.isPresent()) {
+            String errorFeatureName = errorEntry.get().getKey();
```

Comment:
For v1, the existing behavior is that if updateErrors is not empty, we set topLevelError to be none. It seems that we are changing the behavior here?

Reply:
I don't think we are changing it to be none. We set it as usual in line 114.

Reply:
Hmm, we set `topLevelError` to topError in line 100. This means that if there is a feature level error, the top level error is not none, which is different from the current behavior in v1.

Reply:
errorEntry will never be present if version is 1 or below. I agree this is not ideal for readability though.